zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -879,7 +933,19 @@ private void updateNote(NotebookSocket conn, HashSet<String> userAndRoles, Noteb
 
     Note note = notebook.getNote(noteId);
     if (note != null) {
-      boolean cronUpdated = isCronUpdated(config, note.getConfig());
+      if (isCronExecutingUserUpdatePermissionCheckNeeded(
+              notebook.getConf().isNotebookCronExecutingUserLoginUserOnly(),
+              note.getConfig(),
+              config)) {
+        String cronExecutingUser = (String) config.get("cronExecutingUser");
```

Comment:
check that `cronExecutingUser` is a valid value?

Reply:
same in other places

Reply:
@felixcheung I think it is simple to just pass the content of `config.get("cronExecutingUser")` to the `hasCronExecutingUserUpdatePermission` and `cronExecutingUserUpdatePermissionError` methods without validating it at this point.

The case that `cronExecutingUser` is null is considered at `hasCronExecutingUserUpdatePermission` and `cronExecutingUserUpdatePermissionError` methods.

What do you think?